### PR TITLE
native_clock: Re-adjust the RDTSC frequency to its real frequency

### PR DESCRIPTION
**Pull request by Morph1984**
The initial RDTSC frequency, [as determined by CPUID or estimated](https://github.com/yuzu-emu/yuzu/blob/a7792e5ff83523142230951ac7eacbd7685dc40b/src/common/wall_clock.cpp#L58) may not be accurate to its true frequency and may drift relative to the passage of real time.

This PR re-adjusts the RDTSC frequency in accordance to its real-time frequency after a measurement period of 10 seconds, and is done on a separate temporary thread to avoid blocking execution of the main thread.

This was tested to eliminate any remaining time drift on my system¹ in Super Smash Bros. Ultimate², and is also reported to resolve audio drift in Astral Chain, although I am not able to independently verify the latter. Further testing is needed to determine any potential side-effects of this change.

Before:
image
After:
image

As the system clock is used, this batch script can be used to synchronize system time with NTP on Windows (remove the .txt extension).
[time.bat.txt](https://github.com/yuzu-emu/yuzu/files/10915302/time.bat.txt)

¹ The system tested uses an Intel i7-11700K Rocket Lake CPU
² Time drift on Intel Alder Lake/Raptor Lake CPUs may or may not be fixed, I theorize this has to do with potential differences in timing between the P and E cores although I am unable to confirm this. To aid in this I have promoted the thread priorities for the CPU and GPU threads to time critical.

### CODE REVIEW

File: src/common/x64/native_clock.cpp [hunk 1]
```diff
@@ -72,13 +73,29 @@ NativeClock::NativeClock(u64 emulated_cpu_frequency_, u64 emulated_clock_frequen
...
+        const auto end_time = Common::RealTimeClock::Now();
+        const u64 tsc_end = FencedRDTSC();
+        // Calculate differences.
+        const u64 timer_diff = static_cast<u64>(
+            std::chrono::duration_cast<std::chrono::nanoseconds>(end_time - start_time).count());
+        const u64 tsc_diff = tsc_end - tsc_start;
+        const u64 tsc_freq = MultiplyAndDivide64(tsc_diff, 1000000000ULL, timer_diff);
+        rtsc_frequency = tsc_freq;
+        CalculateAndSetFactors();
+    }}.detach();
```
Comment (liamwhite) [inline src/common/x64/native_clock.cpp:94]: `.detach()` with `this` captured causes UB when the object is destroyed. Store the jthread as a member of the NativeClock instead

**Comment by Kelebek1** [comment]
Isn't this affected by the cpu clocking up and down? I thought this was almost un-doable to calculate because of that. Would force clocking the CPU to max help this? Or why not just ditch all this and use a steady clock instead?

**Comment by FernandoS27** [comment]
> Isn't this affected by the cpu clocking up and down? I thought this was almost un-doable to calculate because of that. Would force clocking the CPU to max help this? Or why not just ditch all this and use a steady clock instead?

RTDSC frequency is not affected by overclocking. It remains constant.

**Review by liamwhite — APPROVED** [review]

**Comment by Squall-Leonhart** [comment]
I know this is merged already, but why are you calling RDTSC directly, this is advised against just about everywhere because it doesn't return accurate data, one of the highest search results is even

> [RDTSC only returns evenly numbered values on some chips](https://community.intel.com/t5/Intel-C-Compiler/RDTSC-only-returns-evenly-numbered-values-on-some-chips/td-p/1232792)

**Comment by Squall-Leonhart** [comment]
That must have taken someone alot of coffee and testing.
